File: app/android/app/src/main/java/com/jun/weather/BaseApplication.java
package com.jun.weather;

import android.app.Application;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

import com.jun.weather.repository.AppRepository;
import com.jun.weather.util.CLogger;
import com.jun.weather.util.PreferenceUtils;

public class BaseApplication extends Application {
    private AppExecutor appExecutor;

    @Override
    public void onCreate() {
        super.onCreate();
        this.appExecutor = new AppExecutor();

        setDebug();
        CLogger.init();
        PreferenceUtils.getInstance().init(getBaseContext());
    }

    public AppRepository getRepository() {
        AppRepository appRepository = AppRepository.getInstance(getBaseContext(), appExecutor);
        appRepository.init(this);
        return appRepository;
    }

    public AppExecutor getAppExecutor() {
        return this.appExecutor;
    }

    private static boolean debuggable = false;
    private void setDebug() {
        PackageManager pm = getBaseContext().getPackageManager();
        try {
            //패키지명으로 ApplicationInfo 검색
            ApplicationInfo appInfo = pm.getApplicationInfo(getBaseContext().getPackageName(), 0);
            //ApplicationInfo 의 플래그중 ApplicationInfo.FLAG_DEBUGGABLE set 여부
            debuggable = (0 != (appInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE));
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

    }
    public static boolean isDebug() {
        return debuggable;
    }
}

File: app/android/app/src/main/java/com/jun/weather/BaseApplication.kt
package com.jun.weather

import android.app.Application
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import com.jun.weather.repository.AppRepository
//import com.jun.weather.repository.AppRepository.Companion.getInstance
import com.jun.weather.util.CLogger.Companion.init
import com.jun.weather.util.PreferenceUtils.Companion.getInstance
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

@HiltAndroidApp
class BaseApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        setDebug()
        init()
        getInstance().init(baseContext)
    }

//    val repository = getInstance(baseContext)
    @Inject lateinit var repository: AppRepository

    private fun setDebug() {
        val pm = baseContext.packageManager
        try {
            //패키지명으로 ApplicationInfo 검색
            val appInfo = pm.getApplicationInfo(baseContext.packageName, 0)
            //ApplicationInfo 의 플래그중 ApplicationInfo.FLAG_DEBUGGABLE set 여부
            isDebug = 0 != appInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
    }

    companion object {
        var isDebug = false
            private set
    }
}